src/types.rs

```diff
@@ -583,7 +583,13 @@ impl Rewrite for ast::Ty {
                 let is_dyn = tobj_syntax == ast::TraitObjectSyntax::Dyn;
                 // 4 is length of 'dyn '
                 let shape = if is_dyn { shape.offset_left(4)? } else { shape };
-                let res = bounds.rewrite(context, shape)?;
+                let mut res = bounds.rewrite(context, shape)?;
+                // We may have falsely removed a trailing `+` inside macro call.
+                if context.inside_macro() && bounds.len() == 1 {
```

Comment:
What part of the code removes the `+` character ? Sorry, I don't understand also why `stringify!((foo+));` is considered as a trait object.
Is it possible to have, e.g., `stringify!((foo-));` as well ? If so, it feels like this is a very specific fix that won't resolve other similar instances.

Reply:
ah ok! thanks for the explanation.